src/services/download/mod.rs

```diff
@@ -85,43 +85,42 @@ impl DownloadService {
         source: SourceFileId,
         destination: PathBuf,
     ) -> Box<dyn Future<Item = DownloadStatus, Error = DownloadError> + Send + 'static> {
-        let fut03 = self.worker.spawn(|| async move {
-            match source {
-                SourceFileId::Sentry(source, loc) => {
-                    sentry::download_source(source, loc, destination)
-                        .compat()
-                        .await
-                }
-                SourceFileId::Http(source, loc) => {
-                    http::download_source(source, loc, destination)
-                        .compat()
-                        .await
-                }
-                SourceFileId::S3(source, loc) => {
-                    s3::download_source(source, loc, destination).compat().await
-                }
-                SourceFileId::Gcs(source, loc) => {
-                    gcs::download_source(source, loc, destination)
-                        .compat()
-                        .await
-                }
-                SourceFileId::Filesystem(source, loc) => {
-                    filesystem::download_source(source, loc, destination)
+        let fut03 = self.worker.spawn(
+            "service.download",
+            std::time::Duration::from_secs(3600),
```

Comment:
maybe in the future we want to make this configurable

Reply:
Yeah, I flagged this on the PR that introduced this as fairly arbitrary and probably poorly chosen.

I agree that it should be more controlled by the caller possibly, the parent-most request, the http endpoint invoking this all, should have an overall deadline and all other things that end up being called internally do not need to do any work past this deadline.  In theory cancelling the future is fine, but this breaks across spawn boundaries, and thus service boundaries.  This does indeed mean that the caller needs to supply the deadline at spawn time.

I think it's for another refactor though, the current codebase has various hardcoded timeouts that mostly try to align somewhat with each other.